pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
latest.RESTMapper can tell you that, but it would have to be passed into the validation.  We haven't discussed validating references much beyond this - perhaps filtering out events which don't point to a known object type should happen at a level before validation?

----- Original Message -----
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace",
> >  		event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be
> > normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind ==
> > "Node" {
> 
> I see-- in that case, @ddysher, can you make this code state the distinction
> in those terms? Like make/call a "IsRootResource()" function instead of
> special-casing minion/node.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4396/files#r24872179